helix-core/src/main/java/org/apache/helix/BaseDataAccessor.java

```diff
@@ -43,6 +43,18 @@
    */
   boolean create(String path, T record, int options);
 
+  /**
```

Comment:
General comment:

Good work. This is a very large PR - I think instead of trying to include all changes in one single PR, it would help you and the reviewers to split this up. This will help us:
- Enable modular development
- Facilitate review process (faster turnaround)
- Better quality of reviews

One way I could think of off the top of my head is to first focus on the classes at the lowest level (e.g. ZkConnection, AsyncCallback, etc.) since we need to get those right first. Then move up a layer (ZkClient). And then another layer - (FederatedZkClient, DedicatedZkClient, etc. - these are all similar, so once we get one right, we probably get the rest right). And next, cover BaseDataAccessor and the like.

Reply:
Thank you for the suggestion. I broke it up into multiple PRs. In this one, I just add TTL mode to async create API. In the next, I add it to sync API in `ZkClient` and `ZkConnection`, then to `RealmAwareZkClient` and its descendants, and finally to `BaseDataAcessors`.